src/btree/bt_random.c

```diff
@@ -169,7 +169,8 @@ __wt_row_random_leaf(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
  *	Find a random page in a tree for either sampling or eviction.
  */
 int
-__wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, bool eviction)
+__wt_random_descent(WT_SESSION_IMPL *session,
+    WT_REF **refp, bool eviction, uint32_t additional_flags)
```

Comment:
This method calls `__wt_page_swap`, but without enough context to pass in any cursor flags. I wasn't sure how to funnel the context in here. There are three callers into this method. One is from eviction and the other two are the next/prev implementations for random btree cursors. For this method I've considered:
* Passing in a `cbt` pointer. That seemed inappropriate as `__evict_walk_tree` didn't seem to manage those, so that caller would be passing `null`.
* Passing in a second boolean, `read_once`. From my limited exposure to WT code, this approach did not seem widely adopted when a slightly more flexible API might be useful. I didn't think the small addition to clarity necessarily offset flexibility. Though methods managing two sets of "flag" variables seems uncommon as well.
* Removing the `eviction` boolean and having the callers pass in `flags` altogether. However `eviction` is also used in a couple other spots in the method such that it couldn't entirely be removed.

Reply:
You're right that once we start needing multiple boolean args, what we tend to do is have function flags that then gives flexibility to add new flags in the future too. You're on the right path in your thoughts. My suggestion would be to:
* Remove the `eviction` boolean and only have a new arg `flags`.
* Rename the local `flags` to `page_flags`.
* Callers pass in (via new `flags`) new flags `WT_DESCENT_EVICTION | WT_DESCENT_READ_ONCE`.
* Code then can be:
```
bool eviction;
eviction = LF_ISSET(WT_DESCENT_EVICTION);
if (eviction)
    page_flags = ...;
else
   page_flags = ...;
if (LF_ISSET(WT_DESCENT_READ_ONCE))
    FLD_SET(page_flags, ...);
```

Reply:
Rather than creating a new set of "descent" flags, I'm inclined to use the existing read flags here -- i.e., have callers pass in exactly the flags that will be passed to `__wt_page_swap`.

Also as a general comment, I'd go with `read_flags` as the variable name generally rather than `descent_flags`, `page_swap_flags`, etc.